torch/storage.py

```diff
@@ -156,10 +172,483 @@ def _new_shared(cls, size):
+            return data_str + (
+                f'\n[{torch.typename(self)} with dtype {self.dtype} '
+                f'of size {len(self)}]')
+        else:
+            return data_str + f'\n[{torch.typename(self)} of size {len(self)}]'
+
+    def __repr__(self):
+        return str(self)
+
+    def __iter__(self):
+        return iter(map(lambda i: self[i], range(self.size())))
+
+    def __copy__(self):
+        return self._new_wrapped_storage(copy.copy(self._storage))
+
+    def __deepcopy__(self, memo):
+        return self._new_wrapped_storage(copy.deepcopy(self._storage, memo))
+
+    def __sizeof__(self):
+        return super(TypedStorage, self).__sizeof__() + self.nbytes()
+
+    def clone(self):
+        """Returns a copy of this storage"""
+        return self._new_wrapped_storage(self._storage.clone())
+
+    def tolist(self):
+        """Returns a list containing the elements of this storage"""
+        return list(self)
+
+    def cpu(self):
+        """Returns a CPU copy of this storage if it's not already on the CPU"""
+        return self._new_wrapped_storage(self._storage.cpu())
+
+    def pin_memory(self):
+        """Coppies the  storage to pinned memory, if it's not already pinned."""
+        return self._new_wrapped_storage(self._storage.pin_memory())
+
+    def share_memory_(self):
+        """Moves the storage to shared memory.
+
+        This is a no-op for storages already in shared memory and for CUDA
+        storages, which do not need to be moved for sharing across processes.
+        Storages in shared memory cannot be resized.
+
+        Returns: self
+        """
+        self._storage.share_memory_()
+        return self
+
+    @classmethod
+    def _new_shared(cls, size):
+        """Creates a new storage in shared memory with the same data type"""
+        module = eval(cls.__module__)
+        byte_storage = module.ByteStorage._new_shared(size * cls().element_size())
+        return cls(wrap_storage=byte_storage)
+
+    @property
+    def _cdata(self):
+        return self._storage._cdata
+
+    @property
+    def device(self):
+        return self._storage.device
+
+    def size(self):
+        return len(self)
+
+    def pickle_storage_type(self):
+        try:
+            return _dtype_to_storage_type_map()[self.dtype]
+        except KeyError:
+            raise KeyError(f'dtype {self.dtype} is not recognized')
+
+    def __reduce__(self):
+        b = io.BytesIO()
+        torch.save(self, b, _use_new_zipfile_serialization=False)
+        return (_load_from_bytes, (b.getvalue(),))
+
+    def data_ptr(self):
+        return self._storage.data_ptr()
+
+    def resize_(self, size):
+        self._storage.resize_(size * self.element_size())
+
+    @classmethod
+    def _free_weak_ref(cls, *args, **kwargs):
+        return eval(cls.__module__).ByteStorage._free_weak_ref(*args, **kwargs)
+
+    def _weak_ref(self, *args, **kwargs):
+        return self._storage._weak_ref(*args, **kwargs)
```

Comment:
Not for now, but it would be great to dump as many of these internal methods as possible, by rewiring PyTorch internals that use them to directly work with Storage rather than TypedStorage (when possible)